File: tests/fixtures/fizzbuzz.rs
// FizzBuzz implementation with a deliberate bug for testing
// Bug: Line 9 checks n % 4 instead of n % 5

fn fizzbuzz(n: i32) -> String {
    if n % 15 == 0 {
        "FizzBuzz".to_string()
    } else if n % 3 == 0 {
        "Fizz".to_string()
    } else if n % 4 == 0 {  // BUG: Should be n % 5
        "Buzz".to_string()
    } else {
        n.to_string()
    }
}

fn main() {
    for i in 1..=100 {
        println!("{}: {}", i, fizzbuzz(i));
    }
}

File: tests/fixtures/go/fizzbuzz.go
package main

import "fmt"

// fizzbuzz returns FizzBuzz string for given number
func fizzbuzz(n int) string {
	if n%15 == 0 {
		return "FizzBuzz"
	} else if n%3 == 0 {
		return "Fizz"
	} else if n%5 == 0 {
		return "Buzz"
	} else {
		return fmt.Sprintf("%d", n)
	}
}

func main() {
	fmt.Println("FizzBuzz from 1 to 100:")
	for i := 1; i <= 100; i++ {
		result := fizzbuzz(i)
		fmt.Println(result)
	}
	fmt.Println("Done!")
}
